pkg/api/v1alpha1/cluster_types.go

```diff
@@ -148,14 +155,75 @@ type RegistryMirrorConfiguration struct {
 	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
 }
 
+// OCINamespace represents an entity in a local reigstry to group related images.
+type OCINamespace struct {
+	// Name refers to the name of the upstream registry
+	Registry string `json:"registry,omitempty"`
+	// Namespace refers to the name of a namespace in the local registry
+	Namespace string `json:"namespace,omitempty"`
+}
+
 func (n *RegistryMirrorConfiguration) Equal(o *RegistryMirrorConfiguration) bool {
 	if n == o {
 		return true
 	}
 	if n == nil || o == nil {
 		return false
 	}
-	return n.Endpoint == o.Endpoint && n.Port == o.Port && n.CACertContent == o.CACertContent && n.InsecureSkipVerify == o.InsecureSkipVerify && n.Authenticate == o.Authenticate
+	return n.Endpoint == o.Endpoint && n.Port == o.Port && n.CACertContent == o.CACertContent && n.InsecureSkipVerify == o.InsecureSkipVerify && n.Authenticate == o.Authenticate && OCINamespacesSliceEqual(n.OCINamespaces, o.OCINamespaces)
+}
+
+// GetRegistryMirrorAddressMappings reads OCINamespaces field of RegistryMirrorConfiguration, and returns
+// a mapping from original registry URLs to registry mirror URLs.
+func (n *RegistryMirrorConfiguration) GetRegistryMirrorAddressMappings() map[string]string {
+	registryMirrorMappings := make(map[string]string)
+	base := net.JoinHostPort(n.Endpoint, n.Port)
+	// add registry mirror base address
+	registryMirrorMappings[constants.DefaultRegistryMirrorKey] = base
```

Comment:
This is the key to retrieve the base registry address without any namespace. It is needed for tls and auth sections in containred configuration, because all namespaces under the same registry share the same tls and auth config (https://github.com/containerd/containerd/blob/main/docs/cri/registry.md).